### ISSUE TITLE
Move from python3.4 to newer (and remove support for python 2.7 ?)

### ISSUE THREAD

**Issue by wdevazelhes** [scikit-learn-contrib/metric-learn#166]
As of march 2019, python 3.4 will cease to be maintained. I think we should update the version required (and the one travis checks). 

Since v 0.21, scikit-learn requires python>=3.5. Btw they also don't support python 2.7 anymore. 

To use skggm we should use at least python 3.6.x (see comment https://github.com/metric-learn/metric-learn/pull/162#issuecomment-459733439)

So after all, maybe we could only support python >=3.6.x ? What do you think ?

**Comment by bellet** [comment]
@perimosocordiae and @terrytangyuan probably have a much better informed opinion than myself about this. But indeed, if sklearn stops supporting python 2.7, we probably should as well, and move to 3.6. Since we are preparing a big release the timing is probably right?

**Comment by perimosocordiae** [comment]
I believe numpy and scipy just released their last version supporting python 2, so it's definitely time to start thinking about going 3-only.

How about making v0.5 our last python 2 release? This way any legacy users can still use the new API design.

**Comment by perimosocordiae** [comment]
For the minimum Python 3 version, I think bumping to 3.5 is certainly fine. I'm a little hesitant about going to 3.6, but I haven't seen a recent breakdown of 3.x user frequencies recently so I may be out of date.

Regarding skggm, we can get around version mismatch issues by making it an optional dependency and falling back to sklearn's implementation otherwise.

**Comment by terrytangyuan** [comment]
+1 to drop Python 2 support. It would be better to support 3.5 though if we can (try some workarounds like what @perimosocordiae suggested). If we cannot get around with skggm issue, then 3.6 sounds fine.

**Comment by bellet** [comment]
Agree that making 0.5 the last Python 2 compatible release is good
For Python 3: we could move to 3.4 or 3.5 and raise an error if skgmm is not installed as mentioned by @wdevazelhes ?

sklearn's implementation of graph lasso is definitely not working for us. We will be attending the sklearn sprint later this month though so we can try to work on that then and see if we can get it fixed

**Comment by perimosocordiae** [comment]
I just realized that we haven't dropped python2 support yet. Version 0.5 was almost a year ago now, so I think it's about time we did it. :-)
